### PR TITLE
release(caracal): update snapcraft.yaml

### PR DESCRIPTION
**Pull request by aieri**
Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -44,40 +44,21 @@ parts:
...
-    - git+https://opendev.org/openstack/venus-tempest-plugin.git@3.0.0
-    - git+https://opendev.org/openstack/vitrage-tempest-plugin.git@6.1.1
-    - git+https://opendev.org/openstack/watcher-tempest-plugin.git@2.6.0
-    - git+https://opendev.org/openstack/zaqar-tempest-plugin.git@1.6.0
-    - git+https://opendev.org/openstack/zun-tempest-plugin.git@4.7.1
-    - git+https://opendev.org/openinfra/python-tempestconf.git@3.4.0
+    - git+https://opendev.org/openstack/barbican-tempest-plugin.git@4.0.0
+    - git+https://opendev.org/openstack/cinder-tempest-plugin.git@1.12.0
+    - git+https://opendev.org/openstack/designate-tempest-plugin.git@0.22.0
+    - git+https://opendev.org/openstack/glance-tempest-plugin.git@0.8.0
```
Comment (gabrielcocenza) [inline snap/snapcraft.yaml:52]: Looking at this output I thought that the algorithm would check for every plugin and see if there are two releases and take the last element. Why [glance](https://opendev.org/openstack/releases/src/branch/master/deliverables/caracal/glance-tempest-plugin.yaml) appears here and [cloudkitty](https://opendev.org/openstack/releases/src/branch/master/deliverables/caracal/cloudkitty-tempest-plugin.yaml), [cyborg](https://opendev.org/openstack/releases/src/branch/master/deliverables/caracal/cyborg-tempest-plugin.yaml) and etc are excluded? It's not clear to me

**Review by gabrielcocenza — COMMENTED** [review]
I need some more explanation to understand this PR

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -44,40 +44,21 @@ parts:
     source-type: git
     # 'source-tag' and 'python-packages' are automatically generated and managed by
     # tools/update_snapcraft.py, and therefore, should not be manually modified.
-    source-tag: 34.1.0
+    source-tag: 38.0.0
```
[the remark this answers is not in the document]
Comment (aieri) [reply snap/snapcraft.yaml:47]: right, so `update_snapcraft.py` will always take the latest version from the list

File: snap/snapcraft.yaml (hunk 1 above)
Comment (aieri) [reply snap/snapcraft.yaml:52]: cloudkitty, cyborg, and others are included in the default exclude list: https://github.com/canonical/snap-tempest-automation/blob/main/excluded-plugins.txt

**Review by gabrielcocenza — APPROVED** [review]
LGTM

**Review by chanchiwai-ray — APPROVED** [review]

**Pull request merged**